editoast/src/views/projection.rs

```diff
@@ -555,33 +556,49 @@ impl TrainToProjectOnOperationalPoint {
     }
 }
 
-pub struct OperationalPointProjection(HashMap<String, u64>);
+pub struct OperationalPointProjection(HashMap<OperationalPointIdentifier, u64>);
 
 #[derive(Debug, Error, EditoastError)]
 #[editoast_error(base_id = "operationalPointProjection", default_status = 422)]
 pub enum OperationalPointProjectionError {
     #[error("Expected {expected} distances, but got {found}")]
     InvalidNumberOfDistances { expected: usize, found: usize },
-    #[error("Expected at least two ids")]
-    InvalidNumberOfIds,
+    #[error("Expected at least two refs")]
+    InvalidNumberOfRefs,
 }
 
 impl OperationalPointProjection {
     pub fn new(
-        ids: Vec<String>,
+        op_refs: Vec<OperationalPointIdentifier>,
         distances: Vec<u64>,
+        path_item_cache: &PathItemCache,
     ) -> Result<Self, OperationalPointProjectionError> {
-        if ids.len() < 2 {
-            return Err(OperationalPointProjectionError::InvalidNumberOfIds);
+        if op_refs.len() < 2 {
+            return Err(OperationalPointProjectionError::InvalidNumberOfRefs);
         }
-        if ids.len() != distances.len() + 1 {
+        if op_refs.len() != distances.len() + 1 {
             return Err(OperationalPointProjectionError::InvalidNumberOfDistances {
-                expected: ids.len() - 1,
+                expected: op_refs.len() - 1,
                 found: distances.len(),
             });
         }
+
+        let op_refs = op_refs
```

Comment:
We could add a small comment to explain what is done here.

Reply:
Done